### DISCUSSION TITLE
Is there a way to index or access arbitrary Ogg/Vorbis tags via mpd?

### DISCUSSION THREAD

**Discussion by sjlongland** [MusicPlayerDaemon/MPD#1603] [Q&A] [answered]
Hi all,

I have a large collection of music in Ogg/Vorbis format, curated using custom tools that I have written myself to do tagging and file placement.

In particular, while I try to keep to using "standard" tags, I have some additional non-standard tags that I use… notably:

* `VARIANT` refers to a particular recording of a given artist's song
* `FLAGS` is a comma-separated list of "flags" that indicate common properties of different songs (e.g. `live`, `remastered`, `single`, `extended`… etc)

```
stuartl@vk4msl-ws:~/music/by-track-artist/Queen$ vorbiscomment ../../.by-uuid/4691/896e/bd9d/4a81/8c39/5f01/a2df/50ca.ogg | grep '^\(TITLE\|VARIANT\|FLAGS\)='
TITLE=You’re My Best Friend
VARIANT=Backing Track Mix
stuartl@vk4msl-ws:~/music/by-track-artist/Queen$ vorbiscomment ../../.by-uuid/85ec/0666/6c00/11ea/bfde/acb7/f707/1baa.ogg | grep '^\(TITLE\|VARIANT\|FLAGS\)='
TITLE=You're My Best Friend
stuartl@vk4msl-ws:~/music/by-track-artist/Queen$ vorbiscomment ../../.by-uuid/8690/39cb/e7ea/4860/9593/b504/0811/6ac0.ogg | grep '^\(TITLE\|VARIANT\|FLAGS\)='
FLAGS=Remastered 2011
TITLE=You're My Best Friend
```

(The files are stored under a path derived by UUID to make syncing via `rsync` efficient: the file name never changes even if tags do.  `UUID` is another non-standard tag in the file.)

I'm conscious that MPD seems to have a fixed list of tags it supports: https://mpd.readthedocs.io/en/stable/protocol.html#tags

I was wondering if it were possible to expose those other tags like `VARIANT`?  I recognise that from MPDs perspective, these would just be seen as plain "strings" and it'll be up to other tools I write to bust those apart and process them separately.  I don't need them indexed or searchable at this point, I'd just like to be able to "see" them.

The other alternative of course is I just ignore MPD's library support, and instead create my own that can be used to inject URIs into MPD… which may be the best route forward.  Any solution proposed to https://github.com/MusicPlayerDaemon/MPD/discussions/1236 may work too for this use case, although my application is somewhat different.

**Comment by MaxKellermann** [comment, marked as answer]
The `readcomments` command can be used to download arbitrary Vorbis comments of a song, but you can't search the database by those.
